### PR TITLE
fix dialog aria-expand attribute

### PR DESCRIPTION
**Pull request by okadurin**
Add `aria-expanded` attribute to the dialog element only when it is a non-modal dialog

### CODE REVIEW

File: packages/ui/components/overlays/src/OverlayController.js [hunk 1]
```diff
@@ -640,13 +640,14 @@ export class OverlayController extends EventTarget {
...
       this.__storeOriginalAttrs(this.contentNode, ['role', 'id']);
+      const isModal = this.hasBackdrop;
 
       if (this.invokerNode) {
-        this.__storeOriginalAttrs(this.invokerNode, [
-          'aria-expanded',
-          'aria-labelledby',
-          'aria-describedby',
-        ]);
+        const attributesToStore = ['aria-labelledby', 'aria-describedby'];
```
Comment (gerjanvangeest) [inline packages/ui/components/overlays/src/OverlayController.js:646]: Can we have tests inside the `OverlayController.test.js` for this?

File: packages/ui/components/input-datepicker/test/lion-input-datepicker.test.js [hunk 2]
```diff
@@ -495,11 +495,11 @@ describe('<lion-input-datepicker>', () => {
       const el = await fixture(html`<lion-input-datepicker></lion-input-datepicker>`);
```
Comment (gerjanvangeest) [inline packages/ui/components/input-datepicker/test/lion-input-datepicker.test.js:495]: I am surprised to see that datepicker is a modal dialog, especially when it is presented as local next to the invoker.
On small screens (when it is a bottomsheet), this should be fine, but on desktop size screens it should be a local overlay.

File: packages/ui/components/dialog/test/lion-dialog.test.js [hunk 3]
```diff
@@ -166,4 +166,24 @@ describe('lion-dialog', () => {
       el.setAttribute('opened', '');
     });
   });
+
+  describe('Accessibility', () => {
+    it('adds [aria-expanded] to invoker button', async () => {
```
Comment (gerjanvangeest) [inline packages/ui/components/dialog/test/lion-dialog.test.js:171]:
```suggestion
    it('does not add [aria-expanded] to invoker button', async () => {
```

File: packages/ui/components/overlays/src/OverlayController.js (hunk 1 above)
Comment (okadurin) [reply packages/ui/components/overlays/src/OverlayController.js:646]: Done

File: packages/ui/components/overlays/test/OverlayController.test.js [hunk 4]
```diff
@@ -1528,6 +1528,23 @@ describe('OverlayController', () => {
       expect(ctrl.invokerNode?.getAttribute('aria-expanded')).to.equal('false');
     });
 
+    it('synchronizes [aria-expanded] on invoker when the overlay is modal', async () => {
```
Comment (gerjanvangeest) [inline packages/ui/components/overlays/test/OverlayController.test.js:1531]:
```suggestion
    it('does not synchronize [aria-expanded] on invoker when the overlay is modal', async () => {
```

**Review by gerjanvangeest — APPROVED** [review]

**Pull request merged**